README.md

```diff
@@ -1,27 +1,176 @@
-# broadcast-api
+# Set up an example broadcast API
 
-To install dependencies:
+The **broadcast-example-api** repo provides an example broadcast API to help demonstrate an implementation of a Subscribe button with [built-in consent](https://docs.xmtp.org/consent/subscribe) provided by XMTP.
+
+Follow these steps to set up an example broadcast API like the one backing the example subscribe-broadcast web page [deployed here](https://subscribe-broadcast.vercel.app/subscribe/button) and connected to the XMTP `dev` network.
+
+To get the most out of this repo, use it with the [subscribe-broadcast repo](https://github.com/xmtp/subscribe-broadcast). The subscribe-broadcast repo provides an example subscribe-broadcast web page that you can use with the example broadcast API to send broadcast messages and provide broadcast statuses. To learn how, see [Try the subscribe with built-in consent and broadcast flow](https://docs.xmtp.org/consent/subscribe-broadcast).
+
+**Note**: This project was created using `bun init` in bun v1.0.25. [Bun](https://bun.sh/) is a fast all-in-one JavaScript runtime.
+
+## Install dependencies
 
 ```bash
 bun install
 ```
 
-To run:
+## Get a broadcast key bundle
+
+For this example, you can use this script to get a key bundle for the wallet address you want to use to broadcast the message.
+
+First, pick the wallet address you want to use as the broadcast address. Ensure that it is enabled on the XMTP `dev` network.
+
+### Install the XMTP SDK
+
+If not already installed, install the XMTP JavaScript SDK.
 
 ```bash
-yarn dev
+yarn add @xmtp/xmtp-js ethers
 ```
 
-This project was created using `bun init` in bun v1.0.25. [Bun](https://bun.sh) is a fast all-in-one JavaScript runtime.
-## Example Env
-An example Env File could look something like:
+### Install TSX
+
+If not already installed, install TSX.
+
+```bash
+npm install -g tsx
+```
+
+### Create the script to generate the key bundle
+
+In the `broadcast-example-api` directory, create a `generateAndPrintKeys.ts` file with the following contents.
+
+Replace `<YOUR_PRIVATE_KEY>` with the private key for the wallet address you want to use as the broadcast address.
```

Comment:
I have no idea if there is a better way to get the key bundle - but this is the route ChatGPT took me down. =P Please advise!